### DISCUSSION TITLE
Reference point for standoff centres with respect to sockets

### DISCUSSION THREAD

**Discussion by SimonMerrett** [CRImier/MyKiCad#4] [Q&A] [answered]
Firstly, many thanks for sharing all of this and especially for the write-ups on Hackaday!

I am trying to make a carrier board for M2 and am getting confused over the position of the standoff relative to the socket. In the kindly provided [“all M.2 socket pads” footprint](https://github.com/CRImier/MyKiCad/blob/master/Laptop%20mods/mpcie_nvme/nvme_lib/Conn_TE-M.2-0.5-67P-doublesided.kicad_mod) the 0 datum for the 42, 60, 80 and 110mm standoffs appears to be the centre of the alignment posts but in the only M.2 spec I could find (linked from Timonsku's repo on M.2 connectors) Fig 38 suggests that there is a "card reference plane" 1.75 mm towards the centre of the socket from the centre of the alignment posts: 
![image](https://github.com/CRImier/MyKiCad/assets/26767525/e3a505ea-55d0-45b2-a99c-11e75df7a99a)
Fig. 39 shows with a card in situ:
![image](https://github.com/CRImier/MyKiCad/assets/26767525/ce20434e-0384-4ad3-914f-5803237a1129)

I'm not asking for any changes to anything offered here, just for your insight as someone who is much more experienced in making boards with these connectors on them. Do you think I should calculate the standoff position from the alignment pins or reduce that distance by 1.75 mm?

Thanks for any attention you can give this!

**Comment by CRImier** [comment, marked as answer]
heyo! You're on point - absolutely do reduce the distance by 1.75mm, as in, offset it further towards the center of the socket. I've had to calculate this from datasheets for the AE to M adapter work (is in this repo, converts a 2230 card to 2242), and the M footprint there has an "inserted card edge" line on Fab layer. It's exactly 1.75mm indeed - I didn't refer to the spec to that one, just to the datasheet of the socket I used, but it's been on point as far as I've checked my cards after production, +- 0.5mm. The picture you've posted, has it right too. Hope this helps!

![image](https://github.com/CRImier/MyKiCad/assets/3173633/257e9442-62c7-48a2-af4e-3664dbf7ccce)

Cheers!
Arya

**Reply by SimonMerrett** [reply]
Brill, thanks!
